feat: Add 1.15.0 changelog (#43)
cdr/docs: changelog/1.15.0.md

---
title: "1.15.0"
description: "Released on 01/20/2021"
---

### Breaking Changes ❗

- web: The root dashboard PWA manifest has been removed. Please

>   [uninstall the Coder PWA](https://help.coder.com/hc/en-us/articles/360057764633-Install-the-PWA).

Author: Update this link to point to `pwa.md` on coder/docs

---
title: "1.15.0"
ådescription: "Released on 01/20/2021"
---

### Breaking Changes ❗

> - web: Removed the root dashboard PWA manifest has been removed; uninstall the

Author: ```suggestion
- web: The root dashboard PWA manifest has been removed; uninstall the
```

  Coder PWA. In lieu of this, Coder is moving forward with a PWA-per-application
  approach. To install the Code Server PWA:
  1. Log into Coder and select your environment.
  2. Under **Applications** click **code** to launch code-server.
  3. Follow your browser's instructions for installing the PWA.
- web: Deprecated **Services** in favor of a declarative approach to creating
  environments with YAML called **Environments as Code**. This feature is slated

>   to be released with Coder v. 2.0 (ETA: Q1 of 2021). For now, services will

Author: ```suggestion
  to be released with Coder v2.0 (ETA: Q1 of 2021). For now, services will
```

  continue to function as-is, though this feature will not receive any

>   improvements; they will be removed with Coder v. 2.0

Author: ```suggestion
  improvements; they will be removed with Coder v2.0
```

### Features ✨

- web: Added 30-day notices to dormant accounts listed in the **Manage** >
  **Users** table; this warning indicates these accounts will be removed if not
  reactivated
- web: Added **Cluster Info** page to the Admin Panel. In the future, this page
  will contain the administration and configuration of multiple clusters.
- web: Redesigned Admin Panel
  - Renamed the **General** page to **Infrastructure** and updated page with a
    new design
  - Redesigned the **Authentication**, **Appearance**, **Telemetry** and
    **Dormancy** pages
- web: Changed how Progressive Web Apps function; the code-server application
  can now be installed as a PWA.
- web: Added tooltip with extended information that shows when hovering over the
  Image info in the summary card on the `Environments` page
- web: Redesigned the Coder Dashboard to use NextJS and SSR. Page loads should
  be quicker ⚡.

### Bug Fixes 🐛

- infra: Fixed the removing of query strings from Dev URL authentication
  redirect
- infra: Changed the dotfiles install.sh script so that it is invoked as a child

>   process instead of `exec`'d

Author: ```suggestion
  process instead of `exec`'d. This ensures the remainder of `~/personalize` executes after `install.sh`.
```

- jetbrains: Fixed the appearance of window title and close buttons
- jetbrains: Changed the behavior of links; opening URLS and/or clicking links
  opens a new tab
- jetbrains: Fixed access to the context menu from then Terminal toolbar
- web: Removed display containing environment resources information if the
  environment isn't running
- web: Updated the Import an Image dialog window to hide image tags that have
  already been imported

> - web: Updated the Create/Edit an Environment dialog, the images and tags are

Author: ```suggestion
- web: Updated the Create/Edit an Environment dialog: images and tags are now
```

  [uninstall the Coder PWA](https://help.coder.com/hc/en-us/articles/360057764633-Install-the-PWA).
  Coder is moving forward with a PWA-per-application approach. To install
  `code-server PWA`:
  - click the `Code` application from the _Environments_ page
  - Code Server will open in a new window (or tab) with a web manifest
  - Install the application
  - **Note:** Currently, the PWA scope is limited _per environment_. This means
    each environment requires that the `Code` application is installed as a PWA.
    We recognize that this is not an ideal workflow and as such, it will be
    addressed in a future iteration.
- web: **Services** have been deprecated. We're working hard on a declarative approach
  to creating environments with YAML: **Environments as Code**. This feature is projected
  to land with Coder version 2.0 within Q1 of 2021. For now, **Services** work the
  same as they have before, but are not receiving any improvements in light of the
  upcoming paradigm change. As of 2.0, they'll be completely removed.

### Features ✨

- web: The Manage > Users table displays a notice on dormant accounts that will
  be removed within 30 days

> - web: Admin Panel Cluster Info page has been added

Author: I'll likely be adding more to this line after syncing w Sean

- web: Admin Panel redesigns
  - The "General" page is now "Infrastructure", with a new design
  - Authentication, Appearance, Telemetry and Dormancy have been redesigned
- web: The code-server application can now be installed as a progressive web
  application. Click the code-server icon from the environments page, and then
  follow your browser's instructions for installing the web app. - **NOTE:**
  This application is installed _per environment_. This limitation will be

>   addressed in future iterations.

Author: This is redundant with what's in breaking changes...will consolidate

- web: Hovering an image in the summary card on the `Environments` page elicits
  displays a Tooltip with extended information
- web: We've redesigned the Coder Dashboard to use NextJS and SSR. Page loads should
  be quicker ⚡.

### Bug Fixes 🐛

- infra: Fix removing of query strings from Dev URL authentication redirect
- infra: Dotfiles install.sh script is no longer `exec`'d but instead invoked as

>   a child process.

Author: ```suggestion
  a child process. This ensures the remainder of `~/personalize` executes after `install.sh`.
```

- jetbrains: Fix appearance of window title and close buttons

> - jetbrains: Opening URLS and/or clicking links now open in a new tab

Author: ```suggestion
- jetbrains: Opening URLs and clicking links now open in a new tab
```

---
title: "1.15.0"
ådescription: "Released on 01/20/2021"
---

### Breaking Changes ❗

- web: The root dashboard PWA manifest has been removed; uninstall the
  Coder PWA. In lieu of this, Coder is moving forward with a PWA-per-application
  approach. To install the Code Server PWA:
  1. Log into Coder and select your environment.
  2. Under **Applications** click **code** to launch code-server.
  3. Follow your browser's instructions for installing the PWA.
- web: Deprecated **Services** in favor of a declarative approach to creating
  environments with YAML called **Environments as Code**. This feature is slated
  to be released with Coder v2.0 (ETA: Q1 of 2021). For now, services will
  continue to function as-is, though this feature will not receive any

>   improvements; they will be removed with Coder v2.0

Reviewer: Is the upgrade path from services environments as code? I thought it would've been CVMs in the short term.

Author: This is a valid point, will address

Author: Addressed, checking preview to verify link

Author: Worked!

> - web: The button to create an environment from an image tag has been removed

Reviewer: This is hardly breaking. It would be nice if there was a misc section (or if we just didn't report this at all since it's very minor).

Author: Sure, will remove

### Features ✨

- web: Added 30-day notices to dormant accounts listed in the **Manage** >
  **Users** table; this warning indicates these accounts will be removed if not
  reactivated
- web: Added **Cluster Info** page to the Admin Panel. In the future, this page
  will contain the administration and configuration of multiple clusters.
- web: Redesigned Admin Panel
  - Renamed the **General** page to **Infrastructure** and updated page with a
    new design
  - Redesigned the **Authentication**, **Appearance**, **Telemetry** and
    **Dormancy** pages
- web: Changed how Progressive Web Apps function; the code-server application
  can now be installed as a PWA.
- web: Added tooltip with extended information that shows when hovering over the
  Image info in the summary card on the `Environments` page
- web: Redesigned the Coder Dashboard to use NextJS and SSR. Page loads should

>   be quicker ⚡.

Reviewer: Cluster info tab in admin page landed in this release so we might want to mention it here.

Author: We did - look on line 26
